core/src/main/java/org/apache/accumulo/core/client/admin/TableOperations.java

```diff
@@ -1014,39 +1014,15 @@ default TimeType getTimeType(String tableName) throws TableNotFoundException {
   }
 
   /**
-   * Initiates setting a table to an on-demand state, but does not wait for action to complete
+   * Sets the hosting goal for a range of Tablets in the specified table.
    *
-   * @param tableName the table to set to an on-demand state
-   * @throws AccumuloException when there is a general accumulo error
-   * @throws AccumuloSecurityException when the user does not have the proper permissions
-   * @since 3.1.0
-   */
-  void onDemand(String tableName)
-      throws AccumuloSecurityException, AccumuloException, TableNotFoundException;
-
-  /**
-   * Initiates setting a table to an on-demand state, optionally waits for action to complete
-   *
-   * @param tableName the table to set to an on-demand state
-   * @param wait if true, then will not return until table state is set to an on-demand state
-   * @throws AccumuloException when there is a general accumulo error
-   * @throws AccumuloSecurityException when the user does not have the proper permissions
-   * @since 3.1.0
-   */
-  void onDemand(String tableName, boolean wait)
-      throws AccumuloSecurityException, AccumuloException, TableNotFoundException;
-
-  /**
-   * Check if a table is on-demand through its current goal state only. Could run into issues if the
-   * current state of the table is in between states. If you require a specific state, call
-   * <code>onDemand(tableName, true)</code>, this will wait until the table reaches the desired
-   * state before proceeding.
-   *
-   * @param tableName the table to check if in an on-demand state
-   * @throws AccumuloException when there is a general accumulo error
-   * @return true if table's goal state is online
-   * @since 3.1.0
+   * @param tableName table name
+   * @param range tablet range
+   * @param goal hosting goal
+   * @since ELASTICITY_TODO
    */
-  boolean isOnDemand(String tableName) throws AccumuloException, TableNotFoundException;
-
+  default void setTabletHostingGoal(String tableName, Range range, TabletHostingGoal goal)
```

Comment:
Accumulo APIs that deal with metadata ranges take two Text objects. The range object is a range of row+col. For this operation we only need a row range. If someone sets the columns they would be ignored.  I suppose that since we do not have a RowRange class that is why lots of operations just take two Text objects.

Reply:
Looks like you might have created it already, it's just not a first class citizen.
https://github.com/apache/accumulo/blob/main/core/src/main/java/org/apache/accumulo/core/summary/Gatherer.java#L543

Reply:
I'll move it up and see if I can use it here.